frontend/test/pytest/test_template.py

```diff
@@ -470,7 +778,86 @@ def qft(basis_state):
     params = jnp.array([1, 0, 0])
     interpreted_fn = qml.QNode(qft, device)
     jitted_fn = qjit(interpreted_fn)
-    assert np.allclose(interpreted_fn(params), jitted_fn(params))
+
+    interpreted_result = interpreted_fn(params)
+    jitted_result = jitted_fn(params)
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_aqft(backend):
+    """Test AQFT."""
+
+    def aqft():
+        qml.X(0)
+        qml.Hadamard(1)
+        qml.AQFT(order=1, wires=range(3))
+        return qml.state()
+
+    device = qml.device(backend, wires=3)
+    interpreted_fn = qml.QNode(aqft, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+# Hilbert Schmidt templates take a quantum tape as a parameter.
+# Therefore unsuitable for JIT compilation
```

Comment:
Let's put xfail reasons in the `pytest.xfail("message")` itself, instead of free-floating comments ; )